content/gocourse-2019-05-melbourne.slide

```diff
@@ -813,49 +816,106 @@ Embedding types introduces the possiblity for naming conflicts. Go follows some
 .caption J. D. Frazer "Illiad", User Friendly - [[https://en.wikipedia.org/wiki/User_Friendly#Crud_Puppy][Crud Puppy]]
 
 
-* Errors (1/3)
+* Errors in Go
 
-"Errors are values."  - [[https://go-proverbs.github.io/][Rob Pike]]
-
-The `error` type is a built-in interface:
+- `error` a builtin interface ([[https://golang.org/src/builtin/builtin.go?s=11246:11286#L251][source: error]])
+- Check errors explicitly
+- The [[https://golang.org/pkg/errors/][`errors`]] package has functions for creating new error type
+- Creation of new error type can be done in a few ways:
 
-	type error interface {
-	    Error() string
+	func Sqrt(f float64) (float64, error) {
+		if f < 0 {
+			return 0, errors.New("math: square root of negative number")
+		}
+		// implementation
 	}
+- Or use formatted error message using `fmt.Errorf`:
```

Comment:
To me it's strange to start a phrase with Or. Maybe just write `Use....` or `Alternatively use...`

Reply:
That was a continuation from the previous bullet. I'll check